test/library_test.go

```diff
@@ -0,0 +1,59 @@
+// +build library
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package smoke
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// signal sends a UNIX signal to the test process.
+func signal(s os.Signal) {
```

Comment:
What is the point of these? We should drop them from this repo. They should not be in the hack repo.

Reply:
This is for unit testing the `library.sh` script.